src/blog/2024/05/exploring-node-red-dashboard-2-widgets.md

```diff
@@ -0,0 +1,220 @@
+
+### Displaying data on the chart 
+
+To display charts on the dashboard, we have to use the **ui-chart** widget which allows us to display different types of charts on a dashboard including linear, bar, scatter, etc. This accepts an array and object as input.
+
+1. Drag a **ui-chart** widget onto the canvas.
+2. Double-click on the widget and select Type as **bar**.
+3. Configure the series to **category** and the y-axis to **amount**. This configuration informs the chart that the **amount** property of the input objects will be plotted on the y-axis of the chart.
+4. Connect the output of **join** node named "Combine Objects into Array" to input of the **ui-chart** widget
+
+!["Screenshot displaying the ui-chart widget's configuration"](./images/exploring-dashboard-2-widgets-chart-widget.png "Screenshot displaying the ui-chart widget's configuration"){data-zoomable}
+
+### Adding custom footer with ui-template
+
+With the **ui-template** widget, we can add a custom component to our app using Vue.js. It also allows adding custom CSS for the dashboard and lot of other things. For more information refer to [ui-template docs](https://dashboard.flowfuse.com/nodes/widgets/ui-template.html).
+
+Using this widget, we will add a footer to our application.
+
+1. Drag an **ui-template** widget onto the canvas.
+2. Set the widget type (scoped UI) that will render this widget on the entire dashboard, eliminating the need to add separate footers for each page of the dashboard.
+3. Insert the following vue.js code in the **ui-template** widget. 
+
+```javascript
+<template>
+  <!-- Footer Component -->
+  <div class="footer">
+    <!-- Description of the Income-Expense Tracker -->
+    <div>
+      Welcome to our comprehensive income expense tracker! Take control of your finances by monitoring your income and
+      expenses effortlessly. Our user-friendly interface makes it simple to record transactions, categorize expenses, and
+      analyze your financial trends. With real-time insights into your spending habits, you can make smarter financial
+      decisions and achieve your money goals faster.
+    </div>
+    <!-- Copyright Information -->
+    <div class="copyright">
+      <!-- Display Current Year and Copyright Information -->
+      2024 — <strong>Vuetify</strong>
+    </div>
+  </div>
+</template>
+<style scoped>
+  /* Make the footer occupy all available space */
+  .footer {
+    position:absolute;
+    bottom:0;
+    background-color:rgb(26,26,26);
+    color:rgb(238,238,238);
+    height:130px;
+    text-align:center;
+    padding:14px;
+  }
+
+  .copyright{
+    margin-top:10px;
+  }
+</style>
+```
+
+!["Screenshot displaying the ui-template widget's configuration"](./images/exploring-dashboard-2-widgets-template-widget.png "Screenshot displaying the ui-template widget's configuration"){data-zoomable}
+
+### Deploying your application flow
+
+{% renderFlow %}
```

Comment:
this should be taller, I _think_ you can provide a number as a second argument here, but worth double checking

Reply:
It may be labelled `width`, but it acts as `height`, see the "FUll Flow" example in https://flowfuse.com/blog/2024/04/integrate-dynamodb-with-node-red/

which calls `{% renderFlow 400 %}`

Reply:
No, it's labeled with `height` set to 200 by default. If we wanted to add `width` as well, I would include it.

Reply:
in that case, ignore @ZJvandeWeg and just set it to be taller please :)